base/MAPL_CapGridComp.F90

```diff
@@ -1290,6 +1297,40 @@ subroutine set_grid(this, grid, unusable, rc)
      _RETURN(_SUCCESS)
   end subroutine set_grid
 
+  subroutine inject_external_grid(this, unusable, rc)
+     class(MAPL_CapGridComp),          intent(inout) :: this
+     class(KeywordEnforcer), optional, intent(in   ) :: unusable
+     integer,                optional, intent(  out) :: rc
+
+     type(ESMF_GridMatch_Flag) :: grid_match
+     type(ESMF_Grid)           :: cap_grid,            root_grid
+     logical                   :: cap_grid_is_present, root_grid_is_present
+     integer                   :: status
+
+     _UNUSED_DUMMY(unusable)
+
+     call ESMF_GridCompGet(this%gc, gridIsPresent=cap_grid_is_present, __RC__)
+
+     if (cap_grid_is_present) then
+        call ESMF_GridCompGet(this%gc, grid=cap_grid, __RC__)
```

Comment:
The following logic is a bit obscure to me. If a grid is already set in the root grid comp, should it be validated? Is the assert() necessary? Should we introduce a "force" flag to override previously set grid?
In my opinion, limiting the scope of this method would increase its reliability as you are in control of the conditions under which it is called. However, I defer to your implementation's choice.

Reply:
The assert is there for the case that the root grid comp defines its own grid. In that case, it is safer for the model to fail gracefully, then it is for us to override the defined grid (the grid comp may make assumptions based on its defined grid). Thus it is my opinion that if one is trying to override the root grid comp's defined grid, then one has larger issues to address. 

As far as the validate calls are concerned, we are only calling this method once, so I placed them in there for error catching purposes.

Reply:
The only config where it might be the case that the root has its own grid would be the CTM.   Might be worth checking.

In any event, I consider most of this logic to be "temporary", as I remain committed to evolving MAPL to the point where MAPL components can readily be used as NUOPC components.    But it will take some time to get there.